src/inmanta/data/dataview.py

```diff
@@ -759,6 +760,7 @@ def construct_dtos(self, records: Sequence[Record]) -> Sequence[model.CompileRep
                 exporter_plugin=compile["exporter_plugin"],
                 notify_failed_compile=compile["notify_failed_compile"],
                 failed_compile_message=compile["failed_compile_message"],
+                links=cast(JsonApiLinks, compile.get("links", {})),
```

Comment:
If `JsonApiLinks` were a mapping, I think this case wouldn't be required.

Reply:
Even with the Mapping, I still get mypy errors:
`Argument "links" to "CompileReport" has incompatible type "object"; expected "Mapping[str, str | LinkObject | None]"  [arg-type]
`

Reply:
Ah right, I was talking about the `{}`, but the `compile.get` is even worse than that.